### PR TITLE
Added Site models (#444)

### PR DESCRIPTION
**Pull request by rhysyngsun**
#### What are the relevant tickets?
Part of #444

#### What's this PR do?
Adds a `Site` model and corresponding django-admin ui

#### How should this be manually tested?
Verify you can create a new site in django-admin

### CODE REVIEW

File: open_discussions/views.py [hunk 1]
```diff
@@ -30,19 +31,25 @@ def index(request):
...
         "gaTrackingID": settings.GA_TRACKING_ID,
         "public_path": public_path(request),
         "auth_url": auth_url,
-        "session_url": session_url,
+        "session_url": session_url if site is None else site.session_url,
         "max_comment_depth": settings.OPEN_DISCUSSIONS_MAX_COMMENT_DEPTH,
-        "micromasters_external_login_url": settings.MICROMASTERS_EXTERNAL_LOGIN_URL,
-        "micromasters_base_url": settings.MICROMASTERS_BASE_URL,
+        "micromasters_external_login_url":
+            settings.MICROMASTERS_EXTERNAL_LOGIN_URL if site is None else site.login_url,
```
Comment (noisecapella) [inline open_discussions/views.py]: Would these settings essentially be deprecated after this PR?

**Comment by noisecapella** [comment]
Have you looked at https://docs.djangoproject.com/en/2.0/ref/contrib/sites/?

File: open_discussions/views.py (hunk 1 above)
Comment (rhysyngsun) [reply open_discussions/views.py]: Yeah, actually this slipped through the cracks. I meant to not update these as I don't want complicated code here. And the record for MM won't exist until we create it in django-admin so I want to phase in these changes as outlined in #444. I'll revert this bit of code.

**Comment by rhysyngsun** [comment]
@noisecapella my understanding was django-sites was to support sites that were actually run in the django app itself. Although it brings up an import point, I should rename this model to something like `ExternalSite` to disambiguate.

**Comment by rhysyngsun** [comment]
@noisecapella refactored now and removed the changes to the view. Once we have the model record created in production we can safely just update the view code to either use MM as a default or pull via a key in the JWT token.

**Comment by pdpinch** [comment]
It's not a big deal, but I wonder if this is an issue of semantics. 

Some of the settings here will affect how Discussions functions, so we're customizing the Discussions site. 

But if `ExternalSite` makes more sense to everyone, I'm fine with it.

**Comment by noisecapella** [comment]
Functionality looks great. Just one more thing, is it possible to update the admin site text to say 'MicroMasters login url' and 'MicroMasters base url' instead of just 'login url' and 'base url'?

**Comment by rhysyngsun** [comment]
@noisecapella not sure that makes sense, would it fill in those values from `title`? That's an editable field here.

**Comment by noisecapella** [comment]
It looks like `name` is used for the database field actually... maybe `verbose_name` would work though?

**Comment by rhysyngsun** [comment]
@noisecapella all set

**Pull request merged**